### ISSUE TITLE
tap: xDS and push API

### ISSUE THREAD

**Issue by mattklein123** [envoyproxy/envoy#6110]
Define a tap xDS API that:
1) Streams tap configurations from server to Envoy
2) Streams taps from Envoy to server

**Comment by mattes** [comment]
Thanks @yuval-k for putting in the work to make this API and sink happen. Are you working on the  GRPC sink implementation? Is there anything I can do to help?

**Comment by prestonvanloon** [comment]
I'm interested in the gRPC sink implementation. Has there been any progress?

**Comment by rauljordan** [comment]
Bump on this, interested in having this feature!

**Comment by hnataraj** [comment]
Yes, GRPC streaming of taps is highly desirable. Is this on the radar?

**Comment by prestonvanloon** [comment]
Bump: how can I help contribute to this feature?

**Comment by mattklein123** [comment]
Note that (1) is already handled using ECDS. We just need a gRPC streaming output (2). I will update the title.

**Comment by leosunmo** [comment]
This is something that would be very useful for my org! Is anyone still looking in to this? Anything I can do to help?

**Comment by mattklein123** [comment]
I don't think anyone is working on this, but I would love to see it implemented!

**Comment by leosunmo** [comment]
@abasitt No, nothing on this. I've moved on from the team that needed this.

**Comment by kfaseela** [comment]
@leosunmo : https://groups.google.com/g/envoy-dev/c/ItHvQvODaRk/m/pyBtlR-tAAAJ?utm_medium=email&utm_source=footer

**Comment by kfaseela** [comment]
cc @devannair77

**Comment by devannair77** [comment]
Hi, 
The design doc for [StreamingGrpcSink  ](https://docs.google.com/document/d/1O7Uy4ajjHxU5HQKLQNB5pOBUW35o4kM0/edit?usp=share_link&ouid=115467778777046272644&rtpof=true&sd=true) implementation , Let me know your comments

**Comment by kfaseela** [comment]
> Hi, The design doc for [StreamingGrpcSink ](https://docs.google.com/document/d/1O7Uy4ajjHxU5HQKLQNB5pOBUW35o4kM0/edit?usp=share_link&ouid=115467778777046272644&rtpof=true&sd=true) implementation , Let me know your comments

Thanks @devannair77 , cc @mattklein123 @soulxu

**Comment by mattklein123** [comment]
Can you please make the document into a gdoc that can be commented on? Thank you.

**Comment by kfaseela** [comment]
> Can you please make the document into a gdoc that can be commented on? Thank you.

@mattklein123 I think the doc is already enabled for comments? I do see comments from other people and I am able to add comments myself too.

**Comment by mattklein123** [comment]
Sorry for some reason when I opened it before it opened just the docx file without editing. I see the right link now.

**Comment by mattklein123** [comment]
I left a few comments, thanks.

**Comment by sjtuzbk** [comment]
@devannair77 Is there any progress on this work? I am also very interested in this work.

**Comment by brightsparc** [comment]
@kfaseela I notice that the [StreamingGrpcSink](https://github.com/envoyproxy/envoy/blob/main/source/extensions/common/tap/tap_config_base.cc#L89) is not yet implemented, although the proto is defined.  Is this on the roadmap?

**Comment by kfaseela** [comment]
> @kfaseela I notice that the [StreamingGrpcSink](https://github.com/envoyproxy/envoy/blob/main/source/extensions/common/tap/tap_config_base.cc#L89) is not yet implemented, although the proto is defined.  Is this on the roadmap?

@devannair77 did propose the RFC, but then he got pulled into other priorities that the implementation was not done

**Comment by brightsparc** [comment]
> > @kfaseela I notice that the [StreamingGrpcSink](https://github.com/envoyproxy/envoy/blob/main/source/extensions/common/tap/tap_config_base.cc#L89) is not yet implemented, although the proto is defined.  Is this on the roadmap?
> 
> @devannair77 did propose the RFC, but then he got pulled into other priorities that the implementation was not done

Is there any alternative solution that uses the socket transport that can post to an endpoint, or is that only for streaming results to file/admin endpoint.
